src/main/java/com/openshift/restclient/model/volume/property/IPersistentVolumeProperties.java

```diff
@@ -0,0 +1,10 @@
+package com.openshift.restclient.model.volume.property;
+
+
+import org.jboss.dmr.ModelNode;
+
+public interface IPersistentVolumeProperties {
+
+	void setProperties(ModelNode node);
```

Comment:
We dont want to expose ModelNode on the the interface.  Based on how is consumed,  this is really a copy(IPersistentVolumeProperties) or clone(IPersistentVolumeProperties) method.  I am not fond of casting or type checking either, but please impl a  common base class to the impls (e.g. ModelNodeAdapter)  which would allow minimal casting and type checking to accomplish the same thing without exposing ModelNode